hack/verify-deps.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+set -euo pipefail
+
+function print_failure {
+  echo "There appear to be uncommitted changes to go.mod. Please run 'go tidy' and 'go vendor'."
+  exit 1
+}
+
+go mod vendor
+go mod tidy
```

Comment:
Does this only make sense to do in the CI context?

Reply:
If you were to run it locally, you could catch the problem _before_ you pushed... is there a reason you wouldn't want it to run locally?

Reply:
The script runs `go mod vendor` and `go mod tidy` and then tells the user to run `go mod vendor` and `go mod tidy`.  What would running the same commands a second time accomplish?

Reply:
The instructions are wrong — should it instead say to review the diffs?

Reply:
In the CI context, it makes sense to tell the use to run the command and commit the changes.  In the local-development context, it makes sense to tell the user to review the diff and commit.  (It would be nice if `go mod` had a `--dry-run` flag.)

Reply:
I changed the script to only verify deps when running in a CI environment (see https://steps.svc.ci.openshift.org/help#env)